core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -623,6 +623,9 @@ class SparkContext(config: SparkConf) extends Logging {
 
     // Post init
     _taskScheduler.postStartHook()
+    if (isLocal) {
+      _env.metricsSystem.registerSource(Executor.executorSource)
```

Comment:
So I haven't went and looked but if I don't configure the metrics system files does this cause any overhead?

Reply:
I have not directly measured it, but I'd say the overhead from computing the metrics in the executor namespace is small. Also, we are talking about a change that only affects Spark running in local mode.
However, one important point related to the user-impact of this PR is a consequence of the fact that by default metrics are sunk using the MetricsServelet, which attaches to the REST API this is because of the default value for `"*.sink.servlet.class" = "org.apache.spark.metrics.sink.MetricsServlet"` and `"*.sink.servlet.path" = "/metrics/json"` (see https://spark.apache.org/docs/latest/monitoring.html#metrics ). This means that executor metrics in local mode will be visible under the REST API too.
If we want to be extra cautious about this we can introduce a config (this has been discussed above in this PR and discussion there pointed to not have the extra config).